src/path_manager.c

```diff
@@ -742,28 +752,56 @@ static void handle_mptcp_event(struct l_genl_msg *msg, void *user_data)
 /**
  * @brief Verify that MPTCP is enabled at run-time in the kernel.
  *
- * Mptcpd requires MPTCP to be enabled in the kernel at run-time.
- * Issue a warning if it is not enabled.
+ * Check that MPTCP is supported in the kernel by opening a socket
+ * with the IPPROTO_MPTCP protocol.
+ */
+static bool check_mptcp_socket_support(void)
+{
+#if !HAVE_DECL_IPPROTO_MPTCP
+# define IPPROTO_MPTCP 262
+#endif  // !HAVE_DECL_IPPROTO_MPTCP
+
+        int const fd = socket(AF_INET, SOCK_STREAM, IPPROTO_MPTCP);
+
+        if (fd == -1
+            && errno != EPROTONOSUPPORT
```

Comment:
Do we need to check this errno? I guess it is fine.

But this Python snippet always return the errno 22 (`EINVAL`)

```
import socket
socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM, proto=262)
```

I can get `EPROTONOSUPPORT` with this but it is a different path:

```
import socket
socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM, proto=1)
```

Note that last week, I proposed this change for the upstream MPTCP kernel: https://lists.01.org/pipermail/mptcp/2019-July/001488.html

If MPTCP is disabled,  `socket()` returns -1 and `ENOPROTOOPT` is set. But this patch has not been accepted nor even discussed yet.

Reply:
Thanks again for the review!

I'll drop the `EPROTONOSUPPORT` `errno` check, and just stick with the `EINVAL` documented in the [socket(2)](http://man7.org/linux/man-pages/man2/socket.2.html) man page.